### PR TITLE
1.0 evolution updates

### PR DESCRIPTION
**Pull request by jranson**
udpated docs, helm charts values & configmap, kube configmap, and slight refactor of collapsed forwarding.

### CODE REVIEW

File: deploy/helm/trickster/templates/configmap.yaml [hunk 1]
```diff
@@ -10,129 +10,337 @@ metadata:
...
+
+    [frontend]
     listen_port = {{ .Values.service.port }}
-    {{- if .Values.listenAddress }}
-    listen_address = {{ .Values.listenAddress | quote }}
+    {{- if .Values.frontend }}
+      {{- if .Values.frontend.listenAddress }}
+    listen_address = {{ .Values.frontend.listenAddress | quote }}
+      {{- end }}
+      {{- if .Values.frontend.tlsListenPort }}
```
Comment (LimitlessEarth) [inline deploy/helm/trickster/templates/configmap.yaml]: For the regular `listenPort` we source that information from the service port value. Do we want to do that here as well?

File: deploy/helm/trickster/templates/configmap.yaml [hunk 2]
```diff
@@ -10,129 +10,337 @@ metadata:
...
+    tls_listen_port = {{ .Values.frontend.tlsListenPort }}
+      {{- end }}
+      {{- if .Values.frontend.tlsListenAddress }}
```
Comment (LimitlessEarth) [inline deploy/helm/trickster/templates/configmap.yaml]: Same comment as above on `listen_address`.

File: deploy/helm/trickster/templates/configmap.yaml [hunk 3]
```diff
@@ -10,129 +10,337 @@ metadata:
...
+    {{- if .Values.pingHandlerPath }}
+    ping_handler_path = {{ .Values.pingHandlerPath | quote }}
+    {{- end }}
+
+    [frontend]
     listen_port = {{ .Values.service.port }}
-    {{- if .Values.listenAddress }}
-    listen_address = {{ .Values.listenAddress | quote }}
+    {{- if .Values.frontend }}
+      {{- if .Values.frontend.listenAddress }}
```
[the remark this answers is not in the document]
Comment (jranson) [reply deploy/helm/trickster/templates/configmap.yaml]: We have actually always included `listen_address`, see https://github.com/Comcast/trickster/blob/next/deploy/helm/trickster/templates/configmap.yaml#L15

**Comment by jranson** [comment]
(please do not merge yet)

**Pull request merged**